src/execute/rmd.ts

```diff
@@ -353,3 +371,35 @@ function resolveInlineExecute(code: string) {
     (expr) => `${"`"}r .QuartoInlineRender(${expr})${"`"}`,
   )(code);
 }
+
+export function isKnitrSpinScript(file: string) {
+  const ext = extname(file).toLowerCase();
+  if (ext == ".r") {
+    const text = Deno.readTextFileSync(file);
+    // Consider a .R script that can be spinned if first line starts with a special `#'` comment
+    return /^\s*#'/.test(text);
```

Comment:
Is there something more we could look for? (as I think this would trigger for R code that uses any type of roxygen comment e.g. a package source file or plumber file)

Reply:
Oh indeed. I did not thought package files would be found while rendering with quarto but probably better to be safe here. 

There is no specific spin syntax except `#+` for chunk options but I believe we prefer to use yaml. 

Probably best is requiring a yaml header in the file. First like would need to be `#´ ---`. This would require a user to add one. 

Otherwise we could require a specific fields for quarto. `# /* ` is a special comment that spin will ignore and won't affect output. 
So something like 
```
# /* quarto */
```
but this is quite new and different syntax. 

Maybe requiring yaml header is the simplest here and not too much to ask.